tools/ARIAtools/ARIAProduct.py

```diff
@@ -527,6 +585,166 @@ class (see instructions on how to appropriately add new
+
+        The order of the keys needs to be consistent with the keys in the
+        mappingData function.
+        E.g. a new expected radar-metadata key can be added as XXX to the end
+        of the list "rmdkeys" below, and correspondingly to the end of the list
+        "radarkeys" inside the mappingData function. Same protocol for new
+        expected layer keys in the list "sdskeys" below, and correspondingly in
+        "layerkeys" inside the mappingData function.
+
+        """
+        # initiate variables
+        rdrmetadata_dict = {}
+        sdskeys = ['/science/LSAR/identification/boundingPolygon']
+        #Pass pair name
+        basename     = os.path.basename(fname)
+        self.pairname = basename.split('_')[11][:8] + \
+                                  '_' + basename.split('_')[13][:8]
+                                  
+        # Get polarization
+        pol_dict = {}
+        pol_dict['SV'] = 'VV'
+        pol_dict['SH'] = 'HH'
+        pol_dict['HHNA'] = 'HH'
+        file_pol = pol_dict[basename.split('_')[10]]
+
+        # Radarmetadata names for these versions
+        rdrmetadata_dict['pair_name'] = self.pairname
+        # get mid azimuth time
+        ref_doppler_time = datetime.strptime(basename.split('_')[11],
+            '%Y%m%dT%H%M%S')
+        sec_doppler_time = datetime.strptime(basename.split('_')[12],
+            '%Y%m%dT%H%M%S')
+        mid_dt = ref_doppler_time + \
+                        (sec_doppler_time - ref_doppler_time) / 2
+        mid_datetime_str = mid_dt.strftime('%Y-%m-%dT%H:%M:%S')
+        mid_datetime_str += '.0'
+        rdrmetadata_dict['azimuthZeroDopplerMidTime'] = mid_datetime_str
+
+        # assign latitude to assist with sorting
+        # get product bounding box
+        lyr_pref = '/science/LSAR/GUNW/grids/frequencyA' + \
+                        f'/unwrappedInterferogram/{file_pol}/'
+        proj_location = lyr_pref + 'projection'
+        with h5py.File(fname[8:], 'r') as hdf_gunw:
+            latlon_file_bbox = hdf_gunw[sdskeys[0]]
+            latlon_file_bbox = latlon_file_bbox[()]
+            latlon_file_bbox = loads(latlon_file_bbox)
+        rdrmetadata_dict['centerLatitude'] =  int(latlon_file_bbox.centroid.y)
+        rdrmetadata_dict['projection'] = self.projection
+        pyproj_transformer = Transformer.from_crs('EPSG:4326',
+            f'EPSG:{self.projection}', always_xy=True)
+        file_bbox = transform(lambda x, y, z=None: \
+            pyproj_transformer.transform(x, y), latlon_file_bbox)
+
+        # hardcoded keys
+        rdrmetadata_dict['missionID'] = 'NISAR'
+        rdrmetadata_dict['productType'] = 'UNW GEO IFG'
+        rdrmetadata_dict['wavelength'] = 0.24
+        rdrmetadata_dict['centerFrequency'] = 1.2699997500604727e+09
+        rdrmetadata_dict['slantRangeSpacing'] = 4.461197291666666
+        rdrmetadata_dict['slantRangeStart'] = 727415.98682514
+        rdrmetadata_dict['slantRangeEnd'] = 791384.81843777
```

Comment:
Any reason to hardcode any of these parameters? They all exist in the NISAR GUNW products. The wavelength and center frequency may be different for different modes and should not be assumed constant for all NISAR L-SAR modes.

Reply:
@hfattahi yes, those hardcoded values were a vestige. I went ahead and accessed the appropriate parameters directly.